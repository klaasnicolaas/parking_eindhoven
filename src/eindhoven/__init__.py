"""Asynchronous Python client providing Open Data information of Eindhoven."""

from .eindhoven import ODPEindhoven
from .exceptions import (
    ODPEindhovenConnectionError,
    ODPEindhovenError,
    ODPEindhovenResultsError,
    ODPEindhovenTypeError,
)
from .models import BaseResponse, ParkingSpot, ParkingType

__all__ = [
    "ODPEindhoven",
    "ODPEindhovenConnectionError",
    "ODPEindhovenError",
    "ODPEindhovenResultsError",
    "ODPEindhovenTypeError",
    "BaseResponse",
    "ParkingSpot",
    "ParkingType",
]
