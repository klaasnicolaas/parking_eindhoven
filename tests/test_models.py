"""Test the models."""

from __future__ import annotations

from typing import TYPE_CHECKING

from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from eindhoven import ODPEindhoven, ParkingSpot


async def test_parking_model(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_eindhoven_client: ODPEindhoven,
) -> None:
    """Test the parking model type (1)."""
    aresponses.add(
        "data.eindhoven.nl",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("1_parkings.json"),
        ),
    )
    locations: list[ParkingSpot] = await odp_eindhoven_client.locations(parking_type=1)
    assert locations == snapshot

    # Test the first location geometry properties
    assert locations[0].geometry.latitude == snapshot
    assert locations[0].geometry.longitude == snapshot


async def test_permit_parking_type(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_eindhoven_client: ODPEindhoven,
) -> None:
    """Test the permit parking type (2)."""
    aresponses.add(
        "data.eindhoven.nl",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("2_permit_parkings.json"),
        ),
    )
    locations: list[ParkingSpot] = await odp_eindhoven_client.locations(parking_type=2)
    assert locations == snapshot


async def test_disabled_parking_type(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_eindhoven_client: ODPEindhoven,
) -> None:
    """Test the disabled parking type (3)."""
    aresponses.add(
        "data.eindhoven.nl",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("3_disabled_parkings.json"),
        ),
    )
    locations: list[ParkingSpot] = await odp_eindhoven_client.locations(parking_type=3)
    assert locations == snapshot


async def test_crossed_out_parking_type(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_eindhoven_client: ODPEindhoven,
) -> None:
    """Test the crossed out parking type (4)."""
    aresponses.add(
        "data.eindhoven.nl",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("4_crossed_out_parkings.json"),
        ),
    )
    locations: list[ParkingSpot] = await odp_eindhoven_client.locations(parking_type=4)
    assert locations == snapshot


async def test_loading_parking_type(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_eindhoven_client: ODPEindhoven,
) -> None:
    """Test the load in/out parking type (5)."""
    aresponses.add(
        "data.eindhoven.nl",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("5_loading_parkings.json"),
        ),
    )
    locations: list[ParkingSpot] = await odp_eindhoven_client.locations(parking_type=5)
    assert locations == snapshot


async def test_charging_parking_type(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    odp_eindhoven_client: ODPEindhoven,
) -> None:
    """Test the electric charging parking type (6)."""
    aresponses.add(
        "data.eindhoven.nl",
        "/api/records/1.0/search/",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("6_charging_parkings.json"),
        ),
    )
    locations: list[ParkingSpot] = await odp_eindhoven_client.locations(parking_type=6)
    assert locations == snapshot
