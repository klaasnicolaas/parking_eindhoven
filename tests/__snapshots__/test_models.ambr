# serializer version: 1
# name: test_charging_parking_type
  list([
    ParkingSpot(spot_id='b4b646f4cee515b3b3f3241ad1a7d87c94570fb1', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Professor Dr Dorgelolaan', number=1), geometry=Geometry(coordinates=[5.484090179656347, 51.44487067822449]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='44a0afeaab776825ec60f67b57481df2db6f9f9b', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Biesterweg', number=1), geometry=Geometry(coordinates=[5.487246232687543, 51.42636980073094]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='c9dee06d93fa92847f014e3d18c924cb0e2fc07d', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Torenallee', number=1), geometry=Geometry(coordinates=[5.460958348459787, 51.44543944946517]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='db97183c5f91e586c33f8a4149c1441757b9ea1a', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Wilhelminaplein', number=1), geometry=Geometry(coordinates=[5.470738162332851, 51.43767495234641]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='399c0e7edcbd3f7650740acbb04adc344ec79cef', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Winkelcentrum Woensel', number=1), geometry=Geometry(coordinates=[5.477724256533416, 51.46912251620885]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='d589e7c919c2f0731653061b5b6d4afb2308a456', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Ventoselaan', number=1), geometry=Geometry(coordinates=[5.469230608406298, 51.44031968307088]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='b190bc49f262c5bd26c7abc069fbcf0a5efe8d2a', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Dr Schaepmanlaan', number=1), geometry=Geometry(coordinates=[5.482866882852026, 51.42822819684571]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='64f80f00e9a2887b13d6585d42ee42e678c808e9', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Biesterweg', number=1), geometry=Geometry(coordinates=[5.487240279474211, 51.42634988692747]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='4191e6a36245dcab1bc4ee8bd2dd532823f3e4c1', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Tramstraat', number=1), geometry=Geometry(coordinates=[5.483304267522476, 51.440504884740086]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='89cb75708d8a4abed59a516383d33e0aa69a8ea8', data=ParkingData(parking_type='Parkeerplaats Electrisch opladen', street='Deken van Somerenstraat', number=1), geometry=Geometry(coordinates=[5.4751052452053335, 51.434404968582726]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
  ])
# ---
# name: test_crossed_out_parking_type
  list([
    ParkingSpot(spot_id='d3e6805372ac8e4a1b0f481ea545a04fdf7e2384', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Michelangelolaan', number=1), geometry=Geometry(coordinates=[5.4720713352312655, 51.464376413342755]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='8f67420d70756a916cac9e0919a5e7734c7f0453', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Michelangelolaan', number=1), geometry=Geometry(coordinates=[5.473547800531344, 51.464411039709105]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='5ec8348e0686ac6e2d8d141aaec9a662515ef914', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Bomanshof', number=1), geometry=Geometry(coordinates=[5.485006013818546, 51.430399977668955]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='c3e7439e75e49fc5ee1f0a81cd941b6e6d1922de', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Wilhelminaplein', number=1), geometry=Geometry(coordinates=[5.4713438738447415, 51.43781205618678]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='829bec125aa10e3b5cd19d0f037bc826859da713', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Hof van Eden', number=1), geometry=Geometry(coordinates=[5.4737975043994105, 51.43348816472158]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='e60a33551053bf90eaaccb61f5bef1195afe3438', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Hof van Eden', number=1), geometry=Geometry(coordinates=[5.473617494361154, 51.43337044799295]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='380ffc9c60d5497da0e5b987d3798900a54bf7ae', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Gestelsestraat', number=1), geometry=Geometry(coordinates=[5.472066348597859, 51.43301043342078]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='ae57dfa7636a5cb69b165de4bb881e4a762c9af3', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Raiffeisenstraat', number=1), geometry=Geometry(coordinates=[5.482429392773308, 51.44011918148758]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='d69b299264e5506677a800a7f10b244f52b5c137', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Keizersgracht', number=1), geometry=Geometry(coordinates=[5.475708617669073, 51.4375550012714]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='ec49c1ebb45f11d581508e40afab563e25aa01bc', data=ParkingData(parking_type='Parkeerplaats Afgekruist', street='Professor Dr Dorgelolaan', number=1), geometry=Geometry(coordinates=[5.484752269463942, 51.44435112864605]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
  ])
# ---
# name: test_disabled_parking_type
  list([
    ParkingSpot(spot_id='24482c2bc9a90ad0608a7289758cccd55bf8f5db', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Benedictusstraat', number=1), geometry=Geometry(coordinates=[5.468028294136795, 51.45196590072531]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='084b007fd5476cb134d8900110ee12e7be8e6b24', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Hertog Hendrik van Brabantplein', number=1), geometry=Geometry(coordinates=[5.48576495981318, 51.43258489086503]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='ef77ca1425ceca01f2adf6a99ca2ad1fe5b470a7', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Willem van Hornestraat', number=1), geometry=Geometry(coordinates=[5.487336523369197, 51.43359907711495]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='1c2a3c1d164dd4e6a0ab04221ddc6df5328c4383', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Niet ingevuld', number=1), geometry=Geometry(coordinates=[5.462657655831373, 51.43588190541636]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='16db25a0a1d89b749edd1dfcb30ee64624ab0a16', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Schellingstraat', number=1), geometry=Geometry(coordinates=[5.490066186569297, 51.435898978883735]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='164d1509d6a619595015573ea6ab1b4e96551112', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Winkelcentrum Woensel', number=1), geometry=Geometry(coordinates=[5.477325002022769, 51.4702825688508]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='49bfe27b7e0e4e45e762cebf6781877909182bfc', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Frederiklaan', number=1), geometry=Geometry(coordinates=[5.455993108656147, 51.441731672624805]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='d9c435da2d7ee25526bd69159e972d70c128fed0', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Marconilaan', number=1), geometry=Geometry(coordinates=[5.458827795256831, 51.449891599152984]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='2b9d6e19cee2af5f2ddb03c352d92f8e645a7479', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Winkelcentrum Woensel', number=1), geometry=Geometry(coordinates=[5.478167744280201, 51.469735439493796]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='3052106064129c94c02e620425f7225f66f634b6', data=ParkingData(parking_type='Parkeerplaats Gehandicapten', street='Winkelcentrum Woensel', number=1), geometry=Geometry(coordinates=[5.478165499071164, 51.46969216987122]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
  ])
# ---
# name: test_loading_parking_type
  list([
    ParkingSpot(spot_id='8e9f11a9773f8e7cb7d3cbdd830793e7eee99a77', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Kronehoefstraat', number=1), geometry=Geometry(coordinates=[5.467566134606305, 51.45214043811088]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='6293fb10ddde1ff8e5e0d8a3f1fc92c3628faae5', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Gestelsestraat', number=1), geometry=Geometry(coordinates=[5.4719225803080604, 51.433285852317034]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='aa81a0a51f65e55a0b86f14825d6c5f9574f27a8', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Torenallee', number=1), geometry=Geometry(coordinates=[5.460093640557604, 51.44565424830635]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='9594a960f0e8b660c5fd9facb7bdb67889c91a50', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Grote Berg', number=1), geometry=Geometry(coordinates=[5.4744767226620015, 51.43551302160987]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='8e8aa9d79213f6086792faddd707dbb6ada8a72c', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Philitestraat', number=1), geometry=Geometry(coordinates=[5.457403444276032, 51.448258391184254]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='c767cb8b4ffa660f50d578314106958576c297df', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Hoogstraat', number=1), geometry=Geometry(coordinates=[5.468641608217367, 51.43084001137937]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='e2ff43e1d5b9deb150ce5b581b4919ed06bfb944', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Hoogstraat', number=1), geometry=Geometry(coordinates=[5.46870937352264, 51.43089252074345]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='ae33ef0d248496ee0219c2abc7bc077df05b43f3', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Philitestraat', number=1), geometry=Geometry(coordinates=[5.45682286377073, 51.44901986427776]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='9bf1866eb39da182e04a83c42ce242b94453bc40', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Kronehoefstraat', number=1), geometry=Geometry(coordinates=[5.4674800825964756, 51.45213619642506]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='ac057a4954f9d0e78d708588f1e2b9dda7297f96', data=ParkingData(parking_type='Parkeerplaats laden/lossen', street='Frederiklaan', number=1), geometry=Geometry(coordinates=[5.4560555363217205, 51.44165827018127]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
  ])
# ---
# name: test_parking_model
  list([
    ParkingSpot(spot_id='5943cb52f635a7cb4047bd28415ff3de878ff424', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.474774482175889, 51.42929171364541]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='0f0f4c50bbe1ff96408cdd5d1f6e83531bfe0a5c', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.474524450039031, 51.429371401892176]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='efca03a63e0ecf9def22b843af01f483ace78393', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.474445274023191, 51.42939752664452]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='19e70b93586f1356298a411a2a9acb6ff72fa57e', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.474384856563808, 51.429421031474355]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='e087a727da274abd99dde4d7af07c66001993a6e', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.474132684425932, 51.42947335084988]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='b5812201444522d26adc9697af5791623ed7a63e', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.473815957374367, 51.42956612005954]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='4b601f2fae198a7e090bfe74ae6eb249366b8d71', data=ParkingData(parking_type='Parkeerplaats', street='van Meursstraat', number=1), geometry=Geometry(coordinates=[5.473961684033574, 51.42945262650903]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='5516712e1aa3233e8efd167b54e11674b91fc83b', data=ParkingData(parking_type='Parkeerplaats', street='Potgieterstraat', number=1), geometry=Geometry(coordinates=[5.476749582530006, 51.43162431647239]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='cbbb5bdfcefb0139cdf6a0c7fd9ca2d3616dbd48', data=ParkingData(parking_type='Parkeerplaats', street='Edenstraat', number=1), geometry=Geometry(coordinates=[5.477159477246228, 51.43161227016372]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='32eb8972d9a1c945512c3ab7d41e291ce300aca6', data=ParkingData(parking_type='Parkeerplaats', street='Kronehoefstraat', number=1), geometry=Geometry(coordinates=[5.4665329757278425, 51.4521057145467]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
  ])
# ---
# name: test_parking_model.1
  51.42929171364541
# ---
# name: test_parking_model.2
  5.474774482175889
# ---
# name: test_permit_parking_type
  list([
    ParkingSpot(spot_id='9aff83e308d62b428c81780c99cdafa5085934bd', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Michelangelolaan', number=1), geometry=Geometry(coordinates=[5.471875282928675, 51.464370174138594]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='22dd06a4fc720e412e7899fd99dbd678b09d5247', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Michelangelolaan', number=1), geometry=Geometry(coordinates=[5.473749976284599, 51.464413442384526]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='351f94e6527c748402fa6d2203dbf6cb409c0a37', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Michelangelolaan', number=1), geometry=Geometry(coordinates=[5.47285347272649, 51.464393692077614]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='ac3a3302999f8d0b271a3dd9b48bfaae449814f7', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Michelangelolaan', number=1), geometry=Geometry(coordinates=[5.473425270901313, 51.46440602495267]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='a4064e49ae8a1c4b714b2a2df7250429adbdb6b4', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Smitsstraat', number=1), geometry=Geometry(coordinates=[5.472373282328308, 51.437230597768846]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='f53f1afff40485028639589185d204c253ed025a', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Smitsstraat', number=1), geometry=Geometry(coordinates=[5.472178714735089, 51.43736962937751]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='3f272487df06c055de2c55d27782407a521d796a', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Smitsstraat', number=1), geometry=Geometry(coordinates=[5.472111818477869, 51.43740963285452]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='0ceb8a0083afa92a7d5771c97d1331765e115309', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Smitsstraat', number=1), geometry=Geometry(coordinates=[5.471953747608381, 51.437531514929795]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='05dd0fa03d0e9e3036cb47729c1a478adaa3eeac', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Smitsstraat', number=1), geometry=Geometry(coordinates=[5.471850213455201, 51.43751446685524]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
    ParkingSpot(spot_id='43522f8af90388b303b52f00c4015d5ee5e3b0ec', data=ParkingData(parking_type='Parkeerplaats Vergunning', street='Smitsstraat', number=1), geometry=Geometry(coordinates=[5.471832131954089, 51.437613415634196]), updated_at=datetime.datetime(2024, 4, 1, 21, 45, 5, 786000, tzinfo=datetime.timezone.utc)),
  ])
# ---
